backend/tracim_backend/command/space.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+
+from pyramid.scripting import AppEnvironment
+
+from tracim_backend.command import AppContextCommand
+from tracim_backend.exceptions import ForceArgumentNeeded
+from tracim_backend.exceptions import TracimException
+from tracim_backend.lib.core.workspace import WorkspaceApi
+
+
+class MoveSpaceCommand(AppContextCommand):
+    def get_parser(self, prog_name: str) -> argparse.ArgumentParser:
+        parser = super().get_parser(prog_name)
+        parser.add_argument(
+            "-w",
+            "--workspace_id",
+            help="the workspace id to move",
+            dest="workspace_id",
+            required=True,
+            type=int,
+        )
+        parser.add_argument(
+            "-p",
+            "--new-parent",
+            help="the new parent of workspace id to move",
+            dest="parent_workspace_id",
+            required=False,
+            default=None,
+            type=int,
+        )
+        parser.add_argument(
+            "--force",
+            help="force move of workspace",
+            dest="force",
+            required=False,
+            action="store_true",
+            default=False,
+        )
+        return parser
+
+    def get_description(self) -> str:
+        return """Move a Workspace from a parent to another"""
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        # to not setup object var outside of __init__ .
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+        self._workspace_api = WorkspaceApi(
+            current_user=None, session=self._session, config=self._app_config
+        )
+        if not parsed_args.parent_workspace_id and not parsed_args.force:
+            force_arg_required = (
+                "Warning! You should use --force if you really want to move a space to root."
+            )
+            print(force_arg_required)
+            print("Workspace {} can not be moved to root.".format(parsed_args.workspace_id,))
+            raise ForceArgumentNeeded(force_arg_required)
```

Comment:
What's your reasoning to consider moving to "root" as a special operation needing `--force`? As far as I see it is not destructive (the user can always move back the workspace if he move it to root by accident).

Reply:
The UX "part" of this ticket is not really clear,
my reasoning for this behavior is that:
- it 's possible to make error an set workspace to root
- Currently there is no problem by moving workspace to root, but it may cause issue with a plugin that deal with right related to hierarchy.

I'm open to suggestion for this.

Reply:
You mean the `parent_access` plugin? I just had a look: it adds new members of a workspace to all its parent hierarchy but does nothing when the parent of a workspace is changed.

Reply:
- plugin have been tested, they effectively don't change anything for this.
- i changed the behavior with a 0 value for root instead of force parameter.